tqecjs/src/workspace/index.js

```diff
@@ -137,27 +138,53 @@ export default function TQECApp() {
 	// Add download stim button
 	const downloadStimButton = button('Download Stim file', 100, 50, 'white', 'black');
 	const localTesting = !window.location.href.includes("https://"); // FIXME: this is a hack
-	const stimURL = `${(localTesting ? `http://${testingBackendURL.ip}:${testingBackendURL.port}` : prodBackendURL)}/stim`;
+	const stimURL = `${localTesting ? `http://${testingBackendURL.ip}:${testingBackendURL.port}` : prodBackendURL}/stim`;
 
 	downloadStimButton.on('click', (_e) => {
+		var payload = {plaquettes: []};
+		const tiles = workspace.children.filter((child) => child.isTile);
+		tiles.forEach((tile) => {
+			var _plaquette = {
+				color: tile.plaquette.color.toUint8RgbArray(),
+				qubits: [],
+				layers: []
+			}
+			tile.plaquette.qubits.forEach((qubit) => {
+				assert(qubit.globalX >= 0 && qubit.globalY >= 0, "Qubit coordinates must be non-negative")
+				assert(qubit.qubitType === "data" || qubit.qubitType === "syndrome", "Qubit type must be either 'data' or 'syndrome'")
+				_plaquette.qubits.push({
+					x: qubit.globalX,
```

Comment:
The plaquette qubits should be positioned with respect to the plaquette local coordinate system:
![image](https://github.com/QCHackers/tqec/assets/12374487/d8b41d6e-a42f-4660-b6d2-2618da934690)
If you provide something different, the backend will have to pre-process the coordinates. This is not an issue, we simply have to agree on the convention here.

Reply:
You're right. Let's agree on the convention that the leftmost, uppermost qubit is at point (0,0) and other qubits in the plaquette are positioned relative to that.

Since the plaquette can be any convex hull, we do allow for negative x/y coordinates.

I'll have to think about how I want to implement this, but I think sorting and relabeling in function invoking the `POST` request is a good approach.

Reply:
Following the discussions in #34 I think the local coordinate system will eventually change. Let's keep it like that for the moment, but the origin will likely become the central qubit (labeled 5 in the picture above).